src/Serializer/AbstractItemNormalizer.php

```diff
@@ -349,6 +352,25 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
         return $allowedAttributes;
     }
 
+    /**
+     * {@inheritdoc}
+     */
+    protected function isAllowedAttribute($classOrObject, $attribute, $format = null, array $context = [])
+    {
+        if (!parent::isAllowedAttribute($classOrObject, $attribute, $format, $context)) {
+            return false;
+        }
+
+        $options = $this->getFactoryOptions($context);
+        $propertyMetadata = $this->propertyMetadataFactory->create($context['resource_class'], $attribute, $options);
+        $security = $propertyMetadata->getAttribute('security');
+        if ($this->resourceAccessChecker && $security) {
+            return $this->resourceAccessChecker->isGranted($attribute, $security);
```

Comment:
Shouldn't this be `$context['resource_class']` instead of `$attribute` as first parameter and why don't we pass `object` as `$extraVariables` like in the `DenyAccessListener`?

Reply:
See https://github.com/api-platform/core/pull/3966